src/erldns_zone_cache.erl

```diff
@@ -288,6 +288,17 @@ put_zone_rrset({ZoneName, Digest, Records, _Keys}, RRFqdn, Type, Counter) ->
 		  lager:debug("Not processing RRSet (~p) for Zone (~p): counter (~p) provided is lower than system", [RRFqdn, ZoneName, Counter]);
 	  true -> 
 		  lager:debug("Processing RRSet (~p) for Zone (~p): counter (~p) provided is higher than system", [RRFqdn, ZoneName, Counter]),
+	          % we need to perform extra check to verify that the zone is in cache already as we might
+		  % face timing issue if the initial full zone load for subdomain has not updated the storage yet
+		  % and find_zone_in_cache() will return the parent zone instead
+		  case get_zone(erldns:normalize_name(ZoneName)) of
+			{error,zone_not_found} ->
+			     lager:debug("Zone: ~p not in cache, initiating fetch_zone", [ZoneName]),
+			     erldns_zoneclient:fetch_zone(ZoneName),
```

Comment:
`erldns_zoneclient` is not guaranteed to be present, we cannot call fetch_zone here.

Reply:
It would also create a circular dependency between `erldns` and `erldns_zoneclient`

Reply:
understood - will refactor to address this.